numba/targets/arrayobj.py

```diff
@@ -305,8 +311,10 @@ def iternext_array(context, builder, sig, args, result):
     result.set_valid(is_valid)
 
     with builder.if_then(is_valid):
-        value = _getitem_array1d(context, builder, arrayty, ary, index,
-                                 wraparound=False)
+        value = _getitem_array_int_only(
```

Comment:
Alternative would be to call `_getitem_array_generic` here, which is perhaps marginally slower as it handles negative indices?